### PR TITLE
Qualcomm AI Engine Direct - Adding QNN backend support for reflection_pad1/2d core ATen ops

### PR DESCRIPTION
**Pull request by qti-horodnic**
### Summary
Added support for the aten ops `reflection_pad1d` and `reflection_pad2d`. Both ops use the QNN `pad` op using `mode=reflect` with a decomposition during `to_edge` and extra padding in the `2d` case. The `op_pad` builder was extended to handle these ops.

Note that we cannot currently support `reflection_pad3d` easily, because the input rank for `MIRROR_REFLECT` is [constrained](https://docs.qualcomm.com/doc/80-63442-10/topic/HtpOpDefSupplement.html#constraints-80) to <5 for the QNN op. We can support it by decomposing the op into `2d` slices, but that seems complicated and bug-prone, so I just added implementation for the 1d and 2d `reflection_pad` ops for now.

### Test plan
```
python backends/qualcomm/tests/test_qnn_delegate.py -k TestQNNQuantizedOperator.test_qnn_backend_reflection_pad1d --model SM8750 --host aisw-vm19-labsd --device abf5cb1 --build_folder build-android
python backends/qualcomm/tests/test_qnn_delegate.py -k TestQNNFloatingPointOperator.test_qnn_backend_reflection_pad1d --model SM8750 --host aisw-vm19-labsd --device abf5cb1 --build_folder build-android
python backends/qualcomm/tests/test_qnn_delegate.py -k TestQNNQuantizedOperator.test_qnn_backend_reflection_pad2d --model SM8750 --host aisw-vm19-labsd --device abf5cb1 --build_folder build-android
python backends/qualcomm/tests/test_qnn_delegate.py -k TestQNNFloatingPointOperator.test_qnn_backend_reflection_pad2d --model SM8750 --host aisw-vm19-labsd --device abf5cb1 --build_folder build-android
```
Also ran the existing tests for the `pad` op to make sure nothing was broken with the existing implementation:

```
python backends/qualcomm/tests/test_qnn_delegate.py -k TestQNNQuantizedOperator.test_qnn_backend_pad --model SM8750 --host aisw-vm19-labsd --device abf5cb1 --build_folder build-android
python backends/qualcomm/tests/test_qnn_delegate.py -k TestQNNFloatingPointOperator.test_qnn_backend_pad --model SM8750 --host aisw-vm19-labsd --device abf5cb1 --build_folder build-android
```

### CODE REVIEW

**Review by abhinaykukkadapu — APPROVED** [review]

**Comment by abhinaykukkadapu** [comment]
Thanks for adding support.

**Review by abhinaykukkadapu — APPROVED** [review]

File: backends/qualcomm/_passes/decompose_pad.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+from executorch.exir.dialects._ops import ops as exir_ops
+from executorch.exir.dialects.edge._ops import EdgeOpOverload
+from executorch.exir.pass_base import ExportPass, PassResult
+
+
+class DecomposePad(ExportPass):
```
Comment (abhinaykukkadapu) [inline backends/qualcomm/_passes/decompose_pad.py:13]: Can we add tests to this pass if possible?

File: backends/qualcomm/quantizer/annotators/lpai_rules.py [hunk 2]
```diff
@@ -673,7 +673,14 @@ def annotate(node: Node, quantization_config: QuantizationConfig) -> None:
         annotate_binary(node, quantization_config)
 
 
-@register_annotator([torch.ops.aten.pad.default], QnnConstants.OpPad.op_name)
+@register_annotator(
+    [
+        torch.ops.aten.pad.default,
```
Comment (abhinaykukkadapu) [inline backends/qualcomm/quantizer/annotators/lpai_rules.py:678]: if 	DecomposePad runs in all the transform pipeline, would this hit at all?

File: backends/qualcomm/_passes/decompose_pad.py (hunk 1 above)
Comment (qti-horodnic) [reply backends/qualcomm/_passes/decompose_pad.py:13]: The `DecomposePad` pass is tested implicitly through the e2e tests `test_qnn_backend_pad/reflection_pad_1d/2d`. From what I can see in the codebase, we generally don't add tests for specific op decomposition passes, only for [special passes](https://github.com/pytorch/executorch/blob/main/backends/qualcomm/tests/test_passes.py) like `mha_to_sha`. If you insist I can add an explicit test for it, it's just not really a convention we follow for these kinds of passes.

File: backends/qualcomm/quantizer/annotators/lpai_rules.py (hunk 2 above)
Comment (qti-horodnic) [reply backends/qualcomm/quantizer/annotators/lpai_rules.py:678]: Good question. `transform_for_annotation` calls `QnnPassManager().transform_for_annotation_pipeline()` which runs `DecomposePad` before the quant annotation. When `lpai_rules.py`'s `Pad` annotator runs, it will see `reflection_pad1d/2d` nodes (not `aten.pad.default` with `mode='reflect'`). So, adding `reflection_pad1d/2d` to `lpai_rules.py` is necessary for the LPAI backend to correctly quantize these ops.

File: backends/qualcomm/_passes/decompose_pad.py (hunk 1 above)
Comment (abhinaykukkadapu) [reply backends/qualcomm/_passes/decompose_pad.py:13]: Thanks, we don't have to do it right now, but i think we should harden the testing of passes, runtime etc as part of the CI hardening push. CC: @harshs-qti

Comment (qti-horodnic) [reply backends/qualcomm/_passes/decompose_pad.py:13]: Sure, will make a note of it and sync up internally

Comment (qti-horodnic) [reply backends/qualcomm/_passes/decompose_pad.py:13]: Following up on this, @abhinaykukkadapu FYI that @shewu-quic has a plan to refactor how testing currently works for the QC backend, including testing the decomposition passes explicitly. These changes will be coming in soon
